crates/biome_js_analyze/src/lint/nursery/use_semantic_elements.rs

```diff
@@ -0,0 +1,91 @@
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <div role="checkbox">
+    /// <div role="img">
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// // var a = 1;
+    /// <div>...</div>
+    /// <header>...</header>
+    /// <img alt="" src="image.jpg" />
+    /// ```
+    ///
+    pub UseSemanticElements {
+        version: "next",
+        name: "useSemanticElements",
+        language: "jsx",
+        sources: &[RuleSource::EslintJsxA11y("prefer-tag-over-role")],
+        recommended: true,
+    }
+}
+
+impl Rule for UseSemanticElements {
+    type Query = Aria<JsxOpeningElement>;
+    type State = AnyJsxAttribute;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let attributes = node.attributes();
+        for attr in attributes {
+            let attr_value = attr.as_jsx_attribute().unwrap().name_value_token().unwrap();
+            let attr_name = attr_value.text_trimmed();
+            if attr_name == "role" {
+                let extract_attributes = ctx.extract_attributes(&node.attributes());
+
+                let element: biome_rowan::SyntaxToken<biome_js_syntax::JsLanguage> =
+                    node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+                let element_name = element.text_trimmed();
+                let is_not_interative =
+                    AriaRoles.is_not_interactive_element(element_name, extract_attributes);
+                if is_not_interative {
+                    return Some(attr);
+                }
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        //
+        // Read our guidelines to write great diagnostics:
+        // https://docs.rs/biome_analyze/latest/biome_analyze/#what-a-rule-should-say-to-the-user
+        //
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.range(),
+                markup! {
+                    "This JSX element uses a `role` attribute. Use a semantic element instead."
+                },
+            )
+            .note(markup! {
```

Comment:
I believe we might be able to identify the DOM element that corresponds to that role. We have the metadata for that, but we need a function for that.

If you want, I think this is a nice task to enhance the rule, maybe later with another PR. Let us know what you think.

This is the logic we are interested in:

https://github.com/biomejs/biome/blob/9a05b77cbde0e29135783de2ee1450d3dd5a69ea/crates/biome_aria/src/roles.rs#L1185-L1239

Reply:
@ematipico 
Thank you for telling me.
Actually, I read the code and it seems we can't get the alternative element for the role.

However, it can be big task for me so let me try in the next PR?

Reply:
Yes, that's exactly what I was suggesting